tests/epyccel/recognised_functions/test_numpy_types.py

```diff
@@ -0,0 +1,1234 @@
+
+        assert f_fl_output == test_float_output
+        assert matching_types(f_fl_output, test_float_output)
+
+        f_fl64_output = epyccel_func(fl64)
+        test_float64_output = get_int(fl64)
+
+        assert f_fl64_output == test_float64_output
+        assert matching_types(f_fl64_output, test_float64_output)
+
+    f_fl32_output = epyccel_func(fl32)
+    test_float32_output = get_int(fl32)
+
+    assert f_fl32_output == test_float32_output
+    assert matching_types(f_fl32_output, test_float32_output)
+
+@types('bool[:]')
+@types('int[:]')
+@types('int8[:]')
+@types('int16[:]')
+@types('int32[:]')
+@types('int64[:]')
+@types('float[:]')
+@types('float32[:]')
+@types('float64[:]')
+def get_int64_arr_1d(arr):
+    from numpy import int64, shape
+    a = int64(arr)
+    s = shape(a)
+    return len(s), s[0], a[0], a[1]
+
+@types('bool[:]')
+@types('int[:]')
+@types('int8[:]')
+@types('int16[:]')
+@types('int32[:]')
+@types('int64[:]')
+@types('float[:]')
+@types('float32[:]')
+@types('float64[:]')
+def get_int32_arr_1d(arr):
+    from numpy import int32, shape
+    a = int32(arr)
+    s = shape(a)
+    return len(s), s[0], a[0], a[1]
+
+@types('bool[:]')
+@types('int[:]')
+@types('int8[:]')
+@types('int16[:]')
+@types('int32[:]')
+@types('int64[:]')
+@types('float[:]')
+@types('float32[:]')
+@types('float64[:]')
+def get_int16_arr_1d(arr):
+    from numpy import int16, shape
+    a = int16(arr)
+    s = shape(a)
+    return len(s), s[0], a[0], a[1]
+
+@types('bool[:]')
+@types('int[:]')
+@types('int8[:]')
+@types('int16[:]')
+@types('int32[:]')
+@types('int64[:]')
+@types('float[:]')
+@types('float32[:]')
+@types('float64[:]')
+def get_int8_arr_1d(arr):
+    from numpy import int8, shape
+    a = int8(arr)
+    s = shape(a)
+    return len(s), s[0], a[0], a[1]
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [pytest.mark.fortran]),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="Arrays not handled yet."),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = [
+            pytest.mark.python]
+        )
+    )
+)
+
+@pytest.mark.parametrize( 'get_int', [get_int64_arr_1d, get_int32_arr_1d, get_int16_arr_1d, get_int8_arr_1d])
+def test_numpy_int_array_like_1d(language, get_int):
+
+    size = 5
+
+    bl = randint(0, 1, size=size, dtype= bool)
+
+    integer8 = randint(min_int8, max_int8, size=size, dtype=np.int8)
+    integer16 = randint(min_int16, max_int16, size=size, dtype=np.int16)
+    integer = randint(min_int, max_int, size=size, dtype=np.int)
+    integer32 = randint(min_int32, max_int32, size=size, dtype=np.int32)
+    integer64 = randint(min_int64, max_int64, size=size, dtype=np.int64)
+
+    fl = uniform(min_float / 2, max_float / 2, size = size)
+    fl32 = uniform(min_float32 / 2, max_float32 / 2, size = size)
+    fl32 = np.float32(fl32)
+    fl64 = uniform(min_float64 / 2, max_float64 / 2, size = size)
```

Comment:
Why do we need to divide by 2?

Reply:
i used to use ints boundaries for this type, i think i just missed it because it has no arrays tests, for your question, Python raises `overflow` error when using min/max_float:
```python
>>> import numpy as np
>>> fl = np.random.uniform(np.finfo('float').min, np.finfo('float').max, size = 5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mtrand.pyx", line 1106, in numpy.random.mtrand.RandomState.uniform
OverflowError: Range exceeds valid bounds
>>> fl32 = np.random.uniform(np.finfo('float32').min, np.finfo('float32').max, size = 5)
>>> fl = np.random.uniform(np.finfo('float').min / 2, np.finfo('float').max / 2, size = 5)
```

Reply:
ok :+1: